examples/bri-3/src/bri/identity/bpiSubjects/agents/mockBpiSubjectStorage.agent.ts

```diff
@@ -29,16 +38,22 @@ export class MockBpiSubjectStorageAgent {
     );
 
     this.bpiSubjectsStore.push(createdBp);
-
     return Promise.resolve(createdBp);
   }
 
   async updateBpiSubject(bpiSubject: BpiSubject): Promise<BpiSubject> {
-    const bpToUpdate = this.bpiSubjectsStore.find(
+    const bpToUpdate = this.mapper.map(
+      this.bpiSubjectsStore.find((bp) => bp.id === bpiSubject.id),
```

Comment:
There is a null reference bug here in case this.bpiSubjectStore.find does not return a bpiSubject.

This whole method is strange. We find by id and map to BpiSubject something which is already BpiSubject.
Then we find the index of the same object and use ti to store the changes. I believe we can ditch lines 45, to 49 completely or am i  missing something?

Reply:
No, there is nothing strange happening here. Since we are storing the mock subjects in an array, we need to update the object with the exact value as the received input so that the getter returns the updated value instead of what was there before the update. 

Also, BpiSubject class has variables as {_id, _name,...} so the returned object needs to be mapped to the corresponding {id, name,...} in order for the Object.assign to work.

Reply:
I am not sure i understand. Why would not this work:

`const bpToUpdate = this.bpiSubjectsStore.find(
      (bp) => bp.id === bpiSubject.id,
    );`

`Object.assign(bpToUpdate, bpiSubject);`

`return Promise.resolve(bpToUpdate);`

Both bpToUpdate and BpiSubject are of same type since we are mocking the storage layer.

Reply:
The point that I'm trying to make here is:

The test case has an update call and a getter call. 
As you mentioned, update method returns the Object.assign method.. but it doesn't update the mocked array.
So, the array is still holding the unassigned object.

Reply:
@ognjenkurtic  Actually I gave it a deeper thought, and the reason is the difference between the input and the create BpiSubject keys. Input is {id, name,...} and what's in the array is {_id, _name,...}
I think we can live with this for the test case scenario.

Reply:
Is this becuase we are not mapping from entity mode (Prisma generated model) to our domain object (BpiSubject class) during the fetching from the db? There should always be a mapping in both directions on the level of the storage agent so that the rest of the app talks only the language of our own domain objects.

Reply:
as agreed during the call, the mock generator has been fixed